fanficfare/adapters/adapter_spiritfanfictioncom.py

```diff
@@ -0,0 +1,350 @@
+            if content.text.strip() == 'Gêneros:':
+                parse_until_br('genre', i+1, contents)
+            if content.text.strip() == 'Avisos:':
+                parse_until_br('warnings', i+1, contents)
+
+    ## Normalize chapter URLs in case of title change
+    def normalize_chapterurl(self,url):
+        #https://www.spiritfanfiction.com/historia/story-name-1234/capitulo56
+        url = re.sub(r"https?://("+self.getSiteDomain()+r"/historia/\d+/capitulo\d+)$",
+                     r"https://\1",url)
+        return url
+
+
+    def getChapterText(self, url):
+        logger.debug('Getting chapter text from: %s' % url)
+
+        save_chapter_soup = self.make_soup(self.get_request(url))
+        ## need save_chapter_soup for .new_tag()
+        save_chapter = BeautifulSoup("", "html.parser").new_tag("div")
+
+        chapter_dl_soup = self.make_soup(self.get_request(url))
+        if None == chapter_dl_soup:
+            raise exceptions.FailedToDownload("Error downloading Chapter: %s!  Missing required element!" % url)
+        div_chapter = chapter_dl_soup.find('div', {'class':'clearfix'})
+        chapter_text = div_chapter.find('div', class_='texto-capitulo')
+
+        exclude_notes=self.getConfigList('exclude_notes')
+
+        def decode_email(encoded_email):
+            email = ""
+            r = int(encoded_email[:2], 16)
+            for i in range(2, len(encoded_email), 2):
+                char_code = int(encoded_email[i:i + 2], 16) ^ r
+                email += chr(char_code)
+            return email
+        
+
+        def decode_emails(html_text):
+            # Find all elements with class '__cf_email__'
+            email_elements = html_text.find_all(class_='__cf_email__')
+            for element in email_elements:
+                # Get the data-cfemail attribute value
+                encoded_email = element.get('data-cfemail')
+                if encoded_email:
+                    # Decode the email address
+                    decoded_email = decode_email(encoded_email)
+                    # Replace the obfuscated email with the decoded email
+                    element.string = decoded_email
+            return str(html_text)
+
+
+        def append_tag(elem, tag, string=None, classes=None):
+            '''bs4 requires tags be added separately.'''
+            new_tag = save_chapter_soup.new_tag(tag)
+            if string:
+                new_tag.string=string
+            if classes:
+                new_tag['class']=[classes]
+            elem.append(new_tag)
+            return new_tag
+
+
+        chapimg = chaphead = chapfoot = None
+        # Chapter Image
+        img_url = chapter_text.find('img', {'class':'imagemResponsiva'})
+        if img_url:
+            chapimg = chapter_dl_soup.new_tag('p', style="text-align: center")
+            chapimg.insert(0, chapter_dl_soup.new_tag('img', src=img_url['src']))
+        
+        for tag in chapter_text.find_all('h2'):
```

Comment:
Are you sure this level of reconstructing the chapter text is necessary?  Usually we just include it as is.

Also, again, use `self.make_soup()` or `append_tag()` instead of calling `BeautifulSoup` and `unicode()` instead of `str()`.

Reply:
This is to separate header and footer texts from chapter text, since they are all together.